### ISSUE TITLE
dApp mobile: Make sure WebRTC works as intended

### ISSUE THREAD

**Issue by taleldayekh** [raiden-network/light-client#2600]
## User Story

As Chris I want to be able to rely on WebRTC working for both iOS and Android and if not I want make sure I can fallback on the toDevice.

## Acceptance criteria

WebRTC and toDevice works the same way as in the current dApp.

## Tasks

- [ ] Determine how to with certainty verify that WebRTC and toDevice works as intended.
- [ ] See what needs to be done to include WebRTC in capacitor.

**Comment by andrevmatos** [comment]
Some thoughts for refrence.
Currently, we use some of the RTC classes unconditionally in code, making them a requirement at runtime. We want to remove this requirement, detect dynamically the availability of RTC in the environment and use these at runtime only if available.

One of the places we require it right away is:
https://github.com/raiden-network/light-client/blob/c501569e2b2b48c247cfcaa0e9f9af93088bf693/raiden-ts/src/transport/actions.ts#L39-L44

This `instanceOf` helper is a custom utility from us, and it receives the constructor as parameter and checks the object being validated has the same `constructor.name` as the one for which it was created (we used to use `instanceof` operator, but it makes harder to mock it on the tests, so we switched to the name check). One way to remove this requirement at runtime is to allow this codec to have the class/constructor `name` passed as string literal directly, and an explicit generic parameter to be the validated type (notice that types are always available independently of runtime availability). This way, it'll do the same check, but without requiring the class to be passed at runtime.

Additionally to that, we should move [the `wrtc` polyfill](https://github.com/raiden-network/light-client/blob/c501569e2b2b48c247cfcaa0e9f9af93088bf693/raiden-ts/src/polyfills.ts#L22-L24) to CLI & tests setup instead of generically on the SDK, and to have the `webRTC` capability to [also be dynamically detected and set at runtime](https://github.com/raiden-network/light-client/blob/c501569e2b2b48c247cfcaa0e9f9af93088bf693/raiden-ts/src/epics.ts#L46-L58) depending on the presence of `RTCPeerConnection` in the `globalThis` context (and of course, ensuring [the RTC epic](https://github.com/raiden-network/light-client/blob/c501569e2b2b48c247cfcaa0e9f9af93088bf693/raiden-ts/src/transport/epics/webrtc.ts#L552) is also disabled if the capability is.

This should be enough to ensure the SDK works also on environments without RTC framework natively available or preemptively polyfilled.
